Summarizer API and Gallery API

NAshwinKumar opened the pull request.

NAshwinKumar committed:
Copy modules into src/app/deep-read/core folder

The required modules for extracting text from video and summarization are copied into src/app/deep-read/core folder

NAshwinKumar committed:
Add module for extracting text from video frames

NAshwinKumar committed:
Add APIs for summarization

Currently summarizing only video text
// TODO: Summarize audio text

NAshwinKumar committed:
Add code for extracting text from only slide images

NAshwinKumar committed:
Reorgainze files

NAshwinKumar committed:
Add code to summarize audio(transcript) text

NAshwinKumar committed:
Update requirements

NAshwinKumar committed:
Add destructor for summarizer module

NAshwinKumar committed:
Add code to remove newline characters in summary text

NAshwinKumar committed:
Add CUDA installation script

NAshwinKumar committed:
Add script to download Xception Model

NAshwinKumar committed:
Update .gitignore

NAshwinKumar committed:
Merge branch 'master' into summarization_api

NAshwinKumar committed:
Add code to upload extracted images to Azure blob

NAshwinKumar committed:
Add APIs for Gallery

NAshwinKumar committed:
Update requirements

Acquil commented on src/app/deep-read/.gitignore:
```diff
@@ -144,6 +144,6 @@ model*/
-core/temp/*json
+# Temporary files for transcript and summary
+core/temp/
+!core/temp/README.md
```
Empty files don't get added to git. If we clone and run, there will be an error saying no such directory.

NAshwinKumar replied:
I have added readme.md as an exception in last line, that will make sure that folder is created right?

Acquil replied:
👍

Acquil commented on src/app/deep-read/azure_blob_settings.py:
```diff
@@ -0,0 +1,4 @@
+from os import environ
+CONNECT_STR = environ.get('BLOB_CONNECTION_STR')
```
settings.py can have these variables

Acquil replied:
Done

Acquil commented on src/app/deep-read/core/video_to_text/video_to_text_converter.py:
```diff
@@ -0,0 +1,47 @@
+    
+    #Images are classified as screens or slides(presentation slides) using NN, we will convert only slide images into text
+    def get_slide_images_from_extracted_images(self):
+        self.slide_images = Video_to_Text_Converter.classifier.predict("/media/ashwin/Current/sem_9/open_source/deep-read/src/app/deep-read/core/temp/sample_transcript1")
```
path is absolute here

NAshwinKumar replied:
fixed this bug in [e8f6d24](https://github.com/Acquil/deep-read/pull/32/commits/e8f6d24f2f6d31c30db0df37abe1e4916adab56e)

Acquil commented on src/app/deep-read/cuda_installation_script.sh:
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+# Add NVIDIA package repositories
+wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-repo-ubuntu1804_10.1.243-1_amd64.deb
```
Is there a way to get this instead of specifying 'ubuntu18'?

Acquil reviewed (changes requested):
Needs updates.

Acquil committed:
Refactored code - mcq, gallery and image uploader

NAshwinKumar committed:
Fix bugs

Remove absolute paths in video_to_text_converter module
Reorder the dependencies in requirements(Tensorflow uses older numpy version, so have to install it before pandas)

NAshwinKumar committed:
Decouple summarization and video_to_text_converter APIs

NAshwinKumar committed:
Modify temporary files generation path

Acquil reviewed (approved):
Done

Acquil committed:
Merge branch 'master' into gallery_api
